examples/httprouter/main.go

```diff
@@ -63,7 +63,7 @@ func main() {
 	cfg := certutils.TLSServerConfig{
 		CertPool:    caCerts,
 		BindAddress: "",
-		Port:        8080,
+		Port:        18080,
```

Comment:
Might be worth looking into how  httptest does port finding and assignment.  It seeks for an unused port. Might be a simple thing we can reuse here.